src/filter/subset_obsp/script.py

```diff
@@ -0,0 +1,52 @@
+import mudata as mu
+
+### VIASH START
+par = {
+  'input': 'resources_test/pbmc_1k_protein_v3/pbmc_1k_protein_v3_mms.h5mu',
+  'modality': 'rna',
+  'input_obsp_key': 'distances',
+  'input_obs_key': 'leiden',
+  'input_obs_value': '1',
+  'output_obsm_key': "leiden_1",
+  'output': 'subset_obsp_output.h5mu',
+  'output_compression': None,
+}
+### VIASH END
+
+# START TEMPORARY WORKAROUND setup_logger
+# reason: resources aren't available when using Nextflow fusion
+# from setup_logger import setup_logger
+def setup_logger():
+    import logging
+    from sys import stdout
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+# END TEMPORARY WORKAROUND setup_logger
+logger = setup_logger()
+
+def main():
+    logger.info(f"Reading {par['input']}")
+    mdata = mu.read_h5mu(par["input"])
+    adata = mdata.mod[par["modality"]]
+
+    logger.info(f"Subset columns of obsp matrix under {par['input_obsp_key']} based on {par['input_obs_key']} == {par['input_obs_value']}")
+    obsp = adata.obsp[par["input_obsp_key"]]
+    idx = adata.obs[par["input_obs_key"]].astype(str) == par["input_obs_value"]
+    obsm_subset = obsp[:, idx]
```

Comment:
Just to make sure that my assumption is correct: there is no way to get `keyError` here right? Because obs and obsp share the same indexes? (perhaps add as a comment)

Reply:
Yea correct! I added a comment